src/qml/components/SnapshotSettings.qml

```diff
@@ -0,0 +1,200 @@
+                description: qsTr("You can start using the application more quickly by loading a recent transaction snapshot." +
+                " It will be automatically verified in the background.")
+            }
+
+            ContinueButton {
+                Layout.preferredWidth: Math.min(300, columnLayout.width - 2 * Layout.leftMargin)
+                Layout.topMargin: 40
+                Layout.leftMargin: 20
+                Layout.rightMargin: Layout.leftMargin
+                Layout.bottomMargin: 20
+                Layout.alignment: Qt.AlignCenter
+                text: qsTr("Choose snapshot file")
+                onClicked: fileDialog.open()
+            }
+
+            FileDialog {
+                id: fileDialog
+                folder: shortcuts.home
+                selectMultiple: false
+                selectExisting: true
+                nameFilters: ["Snapshot files (*.dat)", "All files (*)"]
+                onAccepted: {
+                    selectedFile = fileUrl.toString()
+                    snapshotFileName = selectedFile
+                    nodeModel.initializeSnapshot(true, snapshotFileName)
+                }
+            }
+            // TODO: Handle file error signal
+        }
+
+        ColumnLayout {
+            Layout.alignment: Qt.AlignHCenter
+            Layout.preferredWidth: Math.min(parent.width, 450)
+
+            Image {
+                Layout.alignment: Qt.AlignCenter
+                source: "image://images/circle-file"
+
+                sourceSize.width: 200
+                sourceSize.height: 200
+            }
+
+            Header {
+                Layout.fillWidth: true
+                Layout.topMargin: 20
+                Layout.leftMargin: 20
+                Layout.rightMargin: 20
+                header: qsTr("Loading Snapshot")
+                description: qsTr("This might take a while...")
+            }
+
+            // TODO: add progress indicator once the snapshot progress is implemented
+        }
+
+        ColumnLayout {
+            id: loadedSnapshotColumn
+            Layout.alignment: Qt.AlignHCenter
+            Layout.preferredWidth: Math.min(parent.width, 450)
+
+            Image {
+                Layout.alignment: Qt.AlignCenter
+                source: "image://images/circle-green-check"
+
+                sourceSize.width: 60
+                sourceSize.height: 60
+            }
+
+            Header {
+                Layout.fillWidth: true
+                Layout.topMargin: 20
+                headerBold: true
+                header: qsTr("Snapshot Loaded")
+                descriptionBold: false
+                descriptionColor: Theme.color.neutral6
+                descriptionSize: 17
+                descriptionLineHeight: 1.1
+                description: snapshotInfo && snapshotInfo["date"] ?
+                    qsTr("It contains transactions up to %1. Newer transactions still need to be downloaded." +
```

Comment:
It's not really "transactions up to", it's all coins (utxo set) that exist at the snapshot height.

Maybe say something like:

> It contains a snapshot as of block %1. Once loaded, all blocks from %2 to the present are download. After that Bitcoin Core is ready to use, while all historical blocks from genesis to %2 are downloaded and validated in the background.

Reply:
Great suggestion! I will update the description with the next push.

Reply:
In the sequence, this is the fourth snapshot-related screen (state) people see:

1. Snapshot option in settings where people learn about the benefit of this feature
2. "Load snapshot" screen that gives more detail about how it works
3. "Loading snapshot" screen state with a loader that tells people they can do other stuff in the mean time
4. "Snapshot loaded" screen state that informs people what was imported, what to keep in mind and what will happen next (this is the screen that we are talking about in this thread)

I think the point that not all historical transactions are in the snapshot is worth clarifying. The first time people are exposed to what this feature does is in 2, so I'd probably start tweaking at that point. We call it a "recent transaction snapshot" there, which is more precise than just "transactions" ("unspent transaction snapshot" would be a little closer).

We don't need to be 100% technically accurate and I'm not sure users are well-served if we're completely vague about this either. They should be able to form a good enough mental model for making decisions and understanding their impact.

So how about this?

> It contains recent transactions up to %1. Newer and historical transactions will be automatically downloaded and verified.

Or this?

> It contains unspent transactions up to %1. Newer and historical transactions will be automatically downloaded and verified.

Either way, we should ensure to tweak the text across the sequence if we make changes here.

We show the block height in a "View details" option below (did not check if it's implemented in this PR already), so not sure whether we need to include that in the text as well.

As for what happens next, my thinking was that we just say that the data will be filled in and verified here (keep it short), and then have indicators contextually in the UI as to what is happening (balance, send and transaction list have respective load states). So less absorption of logic here, and more contextual info in the appropriate places.

Sorry, this has gotten quite long. Hope it's a helpful response.

Reply:
> We don't need to be 100% technically accurate and I'm not sure users are well-served if we're completely vague about this either. 

I think it's better to be incomplete than incorrect. The nice thing about being vague is that it's clear to the reader there's more to learn.

One reason the phrasing "recent / historical transactions" is potentially confusing is that people may think they can recover historical wallet transactions with it. But it has nothing to do with that. 

The concept that's useful to explain is that the snapshot allows us to start syncing at block %1, saving time by skipping historical blocks. This way the user can get started (with their wallet) earlier. Meanwhile it's safe, because we do check those historical blocks in the background later.

Reply:
TL;DR I reviewed the onboarding text again, and it seems like the use of `transactions` terminology in the snapshot section provides context continuity with what's said before...see screenshot below.

<img width="322" alt="Screenshot 2024-12-13 at 12 45 27" src="https://github.com/user-attachments/assets/06db292a-f5c9-40a5-92c0-43883649467f" />

Unless we come up with something better, I'd be fine with retaining what we have currently though I agree it's inaccurate. Suggestions below:

----------

> I think it's better to be incomplete than incorrect. The nice thing about being vague is that it's clear to the reader there's more to learn.

It does seem like we can strike a better balance between technical accuracy & usefulness to user here, and use vagueness where required.

> We show the block height in a "View details" option below (did not check if it's implemented in this PR already), so not sure whether we need to include that in the text as well.

Agree, this was in the initial mockups I did. (Ideally, I'd like to see these BEFORE the loading process even began as a way of informing the user context about what's about to happen, but after loading is fine too.)

> The first time people are exposed to what this feature does is in 2, so I'd probably start tweaking at that point.

Agree.

> 2. "Load snapshot" screen that gives more detail about how it works.

This screen can omit mention of transactions **"The snapshot allows you to start using the application more quickly.  Once loaded, it will be automatically verified in the background. [Learn more](some_url)"** instead of _"You can start using the application more quickly by loading a recent transaction snapshot. It will be automatically verified in the background."_

> 4. "Snapshot loaded" screen state that informs people what was imported, what to keep in mind and what will happen next (this is the screen that we are talking about in this thread)

I'd suggest something like **"The loaded snapshot contains data up to <date>. The initial download will continue, while the snapshot is verified in the background."** instead of _"It contains transactions up to September 10, 2023.
Newer transactions still need to be downloaded. The data will be verified in the background."_

![Frame 10036](https://github.com/user-attachments/assets/8b72b98a-2c9e-48f0-ab92-c82b238fe93e)

Reply:
Should we move this conversation into a separate issue? It's just about text details, and maybe we don't want to weigh down the overall technical development of this feature? Text is easy to replace separately.

Reply:
@GBKS that makes sense.